src/main/webapp/scripts/index.js

```diff
@@ -2,13 +2,19 @@ $().ready(function() {
   const location = getCookie('location');
   if (location != "") {
     $('#location-checkbox').prop('checked', true);
+    setCoordinateCookie();
+  }
+  else {
+    eraseCookie('coordinates');
   }
 
   $('#location-checkbox').change(function() {
     if(this.checked) {
       setCookie('location', 'true', 1);
+      setCoordinateCookie();
     } else {
       eraseCookie('location');
+      eraseCookie('coordinates');
```

Comment:
I don't think we need the location cookie, we can just check whether coordinates exists.